droidmediabuffer.cpp

```diff
@@ -96,22 +84,25 @@ _DroidMediaBuffer::~_DroidMediaBuffer()
 
 void _DroidMediaBuffer::incRef(struct android_native_base_t* base)
 {
-    DroidMediaBuffer *self = reinterpret_cast<DroidMediaBuffer *>(base);
-    self->m_ref(self->m_data);
+  DroidMediaBuffer * const buffer = reinterpret_cast<DroidMediaBuffer *>(base);
+
+  android_atomic_inc(&buffer->m_refCount);
 }
 
 void _DroidMediaBuffer::decRef(struct android_native_base_t* base)
 {
-    DroidMediaBuffer *self = reinterpret_cast<DroidMediaBuffer *>(base);
-    self->m_unref(self->m_data);
+  DroidMediaBuffer * const buffer = reinterpret_cast<DroidMediaBuffer *>(base);
+
+  if (android_atomic_dec(&buffer->m_refCount) == 1) {
+      delete buffer;
+  }
 }
 
 extern "C" {
 DroidMediaBuffer *droid_media_buffer_create_from_raw_data(uint32_t w, uint32_t h,
```

Comment:
I think this function is no longer used since https://github.com/sailfishos/gst-droid/pull/43 was merged. This might be a good opportunity to drop it.

Reply:
Done.